caluma/caluma_form/structure.py

```diff
@@ -144,6 +144,18 @@ def __init__(self, document, form, question=None, parent=None):
         self.question = question
         self._fields = None
         self._sub_forms = None
+        self._main_form = None
+
+    @property
+    def main_form(self):
+        if self._main_form is None:
+            if hasattr(self.document.family, "work_item") and hasattr(
+                self.document.family.work_item.case.family, "document"
+            ):
```

Comment:
there's probably a cleaner way to do this in python?

Reply:
There's [operator.attrgetter](https://docs.python.org/3/library/operator.html#operator.attrgetter), but it doesn't make things much shorter.

Since the default/fallback is to just have *no value", you could just as well fetch the whole attribute chain and just try/except around it

Reply:
:+1: done